examples/ASDKgram/Sample/CommentFeedModel.m

```diff
@@ -0,0 +1,195 @@
+  if (_refreshFeedInProgress) {
+    return;
+  } else {
+    _refreshFeedInProgress = YES;
+    _currentPage = 0;
+    
+    // FIXME: blow away any other requests in progress
+    
+    [self fetchPageWithCompletionBlock:^(NSArray *newPhotos) {
+      if (block) {
+        block(newPhotos);
+      }
+      _refreshFeedInProgress = NO;
+    } replaceData:YES];
+  }
+}
+
+#pragma mark - Helper Methods
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block
+{
+  [self fetchPageWithCompletionBlock:block replaceData:NO];
+}
+
+- (void)fetchPageWithCompletionBlock:(void (^)(NSArray *))block replaceData:(BOOL)replaceData
+{
+  // early return if reached end of pages
+  if (_totalPages) {
+    if (_currentPage == _totalPages) {
+      return;
+    }
+  }
+  
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    
+    NSMutableArray *newComments = [NSMutableArray array];
+    
+    NSUInteger nextPage = _currentPage + 1;
+    
+    NSString *urlAdditions = [NSString stringWithFormat:@"page=%lu", (unsigned long)nextPage];
+    NSURL *url = [NSURL URLWithString:[_urlString stringByAppendingString:urlAdditions]];
+    NSURLSession *session      = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration ephemeralSessionConfiguration]];
+    NSURLSessionDataTask *task = [session dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+      
+      if (data) {
+        
+        NSDictionary *response = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        
+        if ([response isKindOfClass:[NSDictionary class]]) {
+          
+          _currentPage = [[response valueForKeyPath:@"current_page"] integerValue];
+          _totalPages  = [[response valueForKeyPath:@"total_pages"] integerValue];
+          _totalItems  = [[response valueForKeyPath:@"total_items"] integerValue];
+          
+          NSArray *comments = [response valueForKeyPath:@"comments"];
+          
+          if ([comments isKindOfClass:[NSArray class]]) {
+            
+            NSUInteger numComments = [comments count];
+            if (numComments > NUM_COMMENTS_TO_SHOW) {
+              comments = [comments subarrayWithRange:(NSRange){numComments-NUM_COMMENTS_TO_SHOW, NUM_COMMENTS_TO_SHOW}];
+            }
+            
+            for (NSDictionary *commentDictionary in comments) {
+              
+              if ([response isKindOfClass:[NSDictionary class]]) {
+                
+                CommentModel *comment = [[CommentModel alloc] initWithDictionary:commentDictionary];
+                
+                if (comment) {
+                  [newComments addObject:comment];
+                }
+              }
+            }
+          }
+        }
+      }
+      dispatch_async(dispatch_get_main_queue(), ^{
+        _fetchPageInProgress = NO;
+        if (replaceData) {
+          _comments = [newComments mutableCopy];
```

Comment:
I think you are going to need some form of locking in this class, although it may not be a big issue because I think only one comments request per story is sent in practice...still this code is a lot higher quality than any current project in examples/ !